client/components/contact-details-form-fields/index.jsx

```diff
@@ -0,0 +1,493 @@
+		}
+	}
+
+	handleSubmitButtonClick = event => {
+		event.preventDefault();
+		this.formStateController.handleSubmit( hasErrors => {
+			this.recordSubmit();
+			if ( hasErrors ) {
+				this.focusFirstError();
+				return;
+			}
+			this.props.onSubmit();
+		} );
+	};
+
+	handleFieldChange = ( event ) => {
+		const { name, value } = event.target;
+		const { onFieldChange, contactDetails } = this.props;
+
+		if ( name === 'country-code' ) {
+
+			this.formStateController.handleFieldChange( {
+				name: 'state',
+				value: '',
+				hideError: true,
+			} );
+
+
+			// If the phone number is unavailable, set the phone prefix to the current country
+			if ( value && ! contactDetails.phone ) {
+				this.setState( {
+					phoneCountryCode: value,
+				} );
+			}
+			this.focusAddressField();
+		}
+
+		this.formStateController.handleFieldChange( {
+			name,
+			value,
+		} );
+
+		onFieldChange( this.getMainFieldValues() );
+	};
+
+	handlePhoneChange = ( { value, countryCode } ) => {
+		const { onFieldChange } = this.props;
+
+		this.formStateController.handleFieldChange( {
+			name: 'phone',
+			value,
+		} );
+
+		this.setState( {
+			phoneCountryCode: countryCode,
+		} );
+
+		onFieldChange( this.getMainFieldValues() );
+	};
+
+	getFieldProps = ( name, needsChildRef = false ) => {
+		// if we're referencing a DOM object in a child component we need to add the `inputRef` prop
+		const ref = needsChildRef ? { inputRef: this.getInputRefCallback( name ) } : { ref: name };
+		const { eventFormName } = this.props;
+		const { form } = this.state;
+
+		return {
+			labelClass: 'contact-details-form-fields__label',
+			additionalClasses: 'contact-details-form-fields__field',
+			disabled: formState.isFieldDisabled( form, name ),
+			isError: formState.isFieldInvalid( form, name ),
+			errorMessage: ( formState.getFieldErrorMessages( form, camelCase( name ) ) || [] )
+				.join( '\n' ),
+			onChange: this.handleFieldChange,
+			value: formState.getFieldValue( form, name ) || '',
+			name,
+			eventFormName,
+			...ref,
+		}
+	};
+
+	createField = ( name, componentClass, additionalProps, needsChildRef ) => {
+		return (
+			<div className={ `contact-details-form-fields__container ${ kebabCase( name ) }` }>
+				{ createElement(
+					componentClass,
+					Object.assign(
+						{},
+						{ ...this.getFieldProps( name, needsChildRef ) },
+						{ ...additionalProps }
+					)
+				) }
+			</div>
+		);
+	};
+
+	getCountryCode() {
+		const { form } = this.state;
+		return get( form, 'countryCode.value', '' );
+	}
+
+	renderContactDetailsFields() {
+		const { translate, needsFax, hasCountryStates } = this.props;
+		const countryCode = this.getCountryCode();
+		const { phoneCountryCode } = this.state;
+
+		return (
+			<div className="checkout__domain-details-contact-details-fields">
+				{ this.createField( 'organization', HiddenInput, {
+					label: translate( 'Organization' ),
+					text: translate( "+ Add your organization's name" ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **15** times:
`translate( 'Add Organization Name' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).